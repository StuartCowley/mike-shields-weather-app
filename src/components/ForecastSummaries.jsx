import React from "react";
import PropTypes from "prop-types";
import styles from "../styles/ForecastSummaries.module.css";
import ForecastSummary from "./ForecastSummary";

export default function ForecastSummaries({ forecasts, handleForecastSelect }) {
  return (
    <div
      className={styles["forecast-summaries"]}
      data-testid="forecast-summaries"
    >
      {forecasts.map((forecast) => {
        return (
          <ForecastSummary
            key={`forecast-summary-${date}`}
            date={date}
            description={description}
            temperature={temperature}
            icon={icon}
          />
        );
      })}
    </div>
  );
}

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.string,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
    })
  ).isRequired,
};
