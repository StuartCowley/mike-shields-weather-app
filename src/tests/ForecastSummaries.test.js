import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../components/ForecastSummaries";
import forecast from "../data/forecast.json";

const { forecasts } = forecast;

describe("ForecastSummaries", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={forecasts} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders forecasts as ForecastSummary components", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={forecasts} />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(forecasts.length);
  });
});
