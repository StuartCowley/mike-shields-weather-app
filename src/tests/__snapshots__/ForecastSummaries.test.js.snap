// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
    data-testid="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
      >
        1525046400000
      </h2>
      <p
        class="forecast-summary__description"
      >
        Clear
      </p>
      <p
        class="forecast-summary__icon"
        data-testid="forecast-summary-icon"
      >
        800
      </p>
      <p
        class="forecast-summary__temperature"
      >
        11
      </p>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
      >
        1525132800000
      </h2>
      <p
        class="forecast-summary__description"
      >
        Stormy
      </p>
      <p
        class="forecast-summary__icon"
        data-testid="forecast-summary-icon"
      >
        211
      </p>
      <p
        class="forecast-summary__temperature"
      >
        13
      </p>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
      >
        1525219200000
      </h2>
      <p
        class="forecast-summary__description"
      >
        Heavy Snow
      </p>
      <p
        class="forecast-summary__icon"
        data-testid="forecast-summary-icon"
      >
        602
      </p>
      <p
        class="forecast-summary__temperature"
      >
        1
      </p>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
      >
        1525305600000
      </h2>
      <p
        class="forecast-summary__description"
      >
        Tornado
      </p>
      <p
        class="forecast-summary__icon"
        data-testid="forecast-summary-icon"
      >
        781
      </p>
      <p
        class="forecast-summary__temperature"
      >
        20
      </p>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <h2
        class="forecast-summary__date"
      >
        1525392000000
      </h2>
      <p
        class="forecast-summary__description"
      >
        Hazy
      </p>
      <p
        class="forecast-summary__icon"
        data-testid="forecast-summary-icon"
      >
        721
      </p>
      <p
        class="forecast-summary__temperature"
      >
        25
      </p>
    </div>
  </div>
</DocumentFragment>
`;
